tollway/events.py

```diff
@@ -1,19 +1,27 @@
 import random
+from datetime import datetime, timedelta
 
 from faker import Faker
 from google.pubsub_v1 import PublisherClient, Topic
 
+from tollway.constants import TIME_UNIT, TIMESTAMP_FORMAT
 from tollway.utils import EventsLog, encode_message
 from tollway.vehicle import create_message, create_tollway, create_vehicle
 
 
-def create_late_event(events_log: EventsLog, fake: Faker, tollways: dict) -> dict:
-    past_timestamps = events_log["past_events_timestamps"]
+def _calculate_late_timestamp(time_unit: str, ts: str) -> str:
+    random_integer = random.randint(TIME_UNIT[time_unit]["min"], TIME_UNIT[time_unit]["max"])
+    current_timestamp = datetime.strptime(ts, TIMESTAMP_FORMAT)
+    return (current_timestamp - timedelta(**{time_unit: random_integer})).strftime(TIMESTAMP_FORMAT)
+
+
+def create_late_event(events_log: EventsLog, fake: Faker, tollways: dict, time_unit: str) -> dict:
```

Comment:
when a group of functions repeat args (and grow their number) it could be a hint to refactor into a class, what do you think?

Reply:
I agree. I'm also changing some typing and type hints here so will rework that too.

Reply:
ok cool, maybe log an issue for this?

Reply:
definitely. Just created one.